packages/ckeditor5-image/src/image/imageediting.js

```diff
@@ -44,46 +42,17 @@ export default class ImageEditing extends Plugin {
 	 */
 	init() {
 		const editor = this.editor;
-		const schema = editor.model.schema;
-		const t = editor.t;
 		const conversion = editor.conversion;
 
 		// See https://github.com/ckeditor/ckeditor5-image/issues/142.
 		editor.editing.view.addObserver( ImageLoadObserver );
 
-		// Configure schema.
-		schema.register( 'image', {
-			isObject: true,
-			isBlock: true,
-			allowWhere: '$block',
-			allowAttributes: [ 'alt', 'src', 'srcset' ]
-		} );
-
-		conversion.for( 'dataDowncast' ).elementToElement( {
-			model: 'image',
-			view: ( modelElement, { writer } ) => createImageViewElement( writer )
-		} );
-
-		conversion.for( 'editingDowncast' ).elementToElement( {
-			model: 'image',
-			view: ( modelElement, { writer } ) => toImageWidget( createImageViewElement( writer ), writer, t( 'image widget' ) )
-		} );
-
 		conversion.for( 'downcast' )
 			.add( modelToViewAttributeConverter( 'src' ) )
 			.add( modelToViewAttributeConverter( 'alt' ) )
 			.add( srcsetAttributeConverter() );
 
 		conversion.for( 'upcast' )
-			.elementToElement( {
-				view: {
-					name: 'img',
-					attributes: {
-						src: true
-					}
-				},
-				model: ( viewImage, { writer } ) => writer.createElement( 'image', { src: viewImage.getAttribute( 'src' ) } )
-			} )
```

Comment:
Shouldn't this be in the `ImageBlock` to be able to upcast `img` that is not wrapped with `figure`? Also AFAIR this is used by `viewFigureToModel()` to upcast `img` inside figure.

Reply:
Ok I can see that you moved it to the `getImageTypeMatcher` but this is needed for block images.

Reply:
This is needed only when block image plugin is alone (and then getImageTypeMatcher will return it). With both plugins enabled, `img` not wrapped with `figure` will be upcasted to inline image.